lib/src/main/java/com/tbruyelle/rxpermissions2/RxPermissionsFragment.kt

```diff
@@ -0,0 +1,146 @@
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.Settings
+import android.provider.Settings.canDrawOverlays
+import android.support.annotation.RequiresApi
+import android.support.v4.app.Fragment
+import android.util.Log
+import io.reactivex.subjects.PublishSubject
+import java.util.*
+
+class RxPermissionsFragment : Fragment() {
+
+    // Contains all the current permission requests.
+    // Once granted or denied, they are removed from it.
+    private val mSubjects = HashMap<String, PublishSubject<Permission>>()
+    private var mLogging: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        retainInstance = true
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    fun requestPermissions(permissions: Array<String>) {
+        requestPermissions(permissions, PERMISSIONS_REQUEST_CODE)
+    }
+
+    fun requestSystemPermission(permission: SystemPermission) {
+        val intent = Intent(permission.action, Uri.parse("package:" + requireContext().packageName))
+        startActivityForResult(intent, permission.requestCode)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        SystemPermission.values().find { it.requestCode == requestCode }?.let {
+            val subject = mSubjects[it.permissionName]
+            if (subject == null) {
+                // No subject found
+                Log.e(RxPermissions.TAG, "RxPermissions.onActivityResult invoked but didn't find the corresponding permission request.")
+                return
+            }
+            mSubjects.remove(it.permissionName)
+            val granted = it.enabledChecker.invoke(requireContext())
+            subject.onNext(Permission(it.permissionName, granted, false))
+            subject.onComplete()
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+
+        if (requestCode != PERMISSIONS_REQUEST_CODE) return
+
+        val shouldShowRequestPermissionRationale = BooleanArray(permissions.size)
+
+        for (i in permissions.indices) {
+            shouldShowRequestPermissionRationale[i] = shouldShowRequestPermissionRationale(permissions[i])
+        }
+
+        onRequestPermissionsResult(permissions, grantResults, shouldShowRequestPermissionRationale)
+    }
+
+    fun onRequestPermissionsResult(permissions: Array<String>, grantResults: IntArray, shouldShowRequestPermissionRationale: BooleanArray) {
+        var i = 0
+        val size = permissions.size
+        while (i < size) {
```

Comment:
The previous code had this as a for loop,
let's put this back as a for loop. Weird that the Kotlin converter made a it a while loop